src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
+    applies a correction related to the number of ensemble members and the number of
+    ensemble members that predict the event.
+
+    The fair Brier score for ensembles is defined for a single event threshold as:
+
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2 - \\frac{i(m-i)}{m^2(m-1)}
+
+    Where:
+
+    - :math:`i` is the number of ensemble members that predict the event
+    - :math:`m` is the total number of ensemble members
+    - :math:`y` is the observed event
+
+    When the fair correction is not applied (i.e., ``fair_correction=False``),
+    the Brier score is calculated as:
+
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2
+
+    If ``fair_correction`` is set to ``True`` and the number of ensemble members is 1, the
+    fair correction will not be applied to avoid division by zero. If you want to
+    set the minimum number ensemble members required to calculate the Brier score,
+    we recommend preprocessing your data to remove data points with fewer than the
+    minimum number of ensemble members that you want.
+
+    Args:
+        fcst: Forecast or predicted variables in xarray.
+        obs: Observed variables in xarray.
+        ensemble_member_dim: The dimension name of the ensemble member.
+        thresholds: The threshold(s) to use for the Brier score that define what is
+            considered an event. If multiple thresholds are provided, they should
+            be monotonically increasing with no NaNs
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the Brier score. All other dimensions will be preserved.
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the Fair Brier score. All other dimensions will be reduced. As a
+            special case, 'all' will allow all dimensions to be preserved. In
+            this case, the result will be in the same shape/dimensionality
+            as the forecast, and the errors will be the absolute error at each
+            point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom).
+        fair_correction: Whether or not to apply the fair Brier score correction. Default is True.
+        threshold_mode: The mode to use when determining threshold exceedance. Default is '>='
+            which means that the threshold is exceeded if the forecast is greater than or equal
+            to the threshold. Alternatively, you can provide ``threshold_mode='>'` which means
+            that the threshold is exceeded if the forecast is strictly greater than the threshold.
+
+
+    Returns:
+        The Brier score for the ensemble forecast.
+
+    Raises:
+        ValueError: if values in ``thresholds`` are not monotonically increasing.
```

Comment:
an alternative is that the user supplies `threshold_dim` with `threshold_dim='threshold'` by default